src/transformers/models/deberta/modeling_deberta.py

```diff
@@ -508,16 +565,19 @@ def build_relative_position(query_size, key_size, device):
     P_k\\)
 
     Args:
-        query_size (int): the length of query
-        key_size (int): the length of key
+        query_layer (Tensor): the query layer or tensor from which to infer size and device
+        key_layer (Tensor): the key layer or tensor from which to infer size and device
 
     Return:
-        `torch.LongTensor`: A tensor with shape [1, query_size, key_size]
+        `torch.LongTensor`: A tensor with shape [1, query_layer.size, key-layer.size]
 
     """
 
-    q_ids = torch.arange(query_size, dtype=torch.long, device=device)
-    k_ids = torch.arange(key_size, dtype=torch.long, device=device)
+    query_size = query_layer.size(-2)
+    key_size = key_layer.size(-2)
+
+    q_ids = torch.arange(query_size, dtype=torch.long, device=query_layer.device)
+    k_ids = torch.arange(key_size, dtype=torch.long, device=key_layer.device)
     rel_pos_ids = q_ids[:, None] - k_ids.view(1, -1).repeat(query_size, 1)
     rel_pos_ids = rel_pos_ids[:query_size, :]
```

Comment:
```suggestion
    rel_pos_ids = rel_pos_ids[q_ids, :]
```
pretty sure when compiling this is faster. I don't know exactly for scripting, but it's slow for cpu infer. Just noting here !